packages/client-core/src/networking/NetworkInstanceProvisioning.tsx

```diff
@@ -236,14 +240,30 @@ export const PartyInstanceProvisioning = () => {
   return null
 }
 
+export const WorldNetworkPeer = (props: { userID: UserId; name: string }) => {
+  useEffect(() => {
+    if (props.userID === Engine.instance.userId) return
+    NotificationService.dispatchNotify(`${props.name} ${t('common:toast.joined')}`, { variant: 'default' })
+    return () => {
+      NotificationService.dispatchNotify(`${props.name} ${t('common:toast.left')}`, { variant: 'default' })
+    }
+  }, [])
+
+  return <></>
+}
+
 export const InstanceProvisioning = () => {
   const networkConfigState = useHookstate(getMutableState(NetworkState).config)
+  const userNames = useHookstate(getMutableState(WorldState).userNames)
```

Comment:
Usernames are not the same as connected peers or users; can we use a more appropriate source of truth?

Reply:
We should be getting a list of the connected user ids, and mapping those into WorldNetworkUser reactors

Reply:
WorldState is derived from connected peers, it is a mapping of connected users to their names, this is the source of truth for this information.

Reply:
The usernames map should be a source of truth for mapping user ids to a user name, and nothing more. Otherwise we are overloading it to represent networking state, and that's prone to error.

Reply:
Alternatively, we can put the username map elsewhere, and just have connectedUserIds in the NetworkState